openproject/client.py

```diff
@@ -0,0 +1,131 @@
+import httpx
+from exceptions import APIError, AuthenticationError
+
+
+class Client:
+    def __init__(self, base_url: str, api_token: str):
+        self.base_url = base_url
+        self.api_version = "v3"
+        self.api_token = api_token
+
+    def _handle_response(self, response: httpx.Response):
+        if response.status_code == 401:
+            raise AuthenticationError(response.json()["message"], response.status_code)
+
+        if response.status_code == 204:
+            return None
+
+        try:
+            response.raise_for_status()
+
+            return response.json()
+        except:
+            error = f"{response.text}"
+            if "application/json" in response.headers["Content-Type"]:
```

Comment:
Let's make this a bit more resilient by:

- not requiring the header to be present
- making it case insensitive

```suggestion
            content_type = response.headers.get("Content-Type", "").lower()
            if content_type == "application/json":
```